From 2b12ace819ccccffe99106f84090c76f7fa85acc Mon Sep 17 00:00:00 2001
From: Tomas Winkler <tomas.winkler@intel.com>
Date: Thu, 31 May 2018 23:20:25 +0300
Subject: [PATCH 1708/2302] Revert "char: rpmb: Document Replay Protected
 Memory Block (RPMB) subsystem"

This reverts commit 6d4c38bf102726962d0eefd2ccf587ac1985f0ea.

Change-Id: Ibaaa35a041689797452a318aa610c3eb9f23fd37
---
 Documentation/conf.py                    |   2 -
 Documentation/index.rst                  |   1 -
 Documentation/rpmb/conf.py               |   5 --
 Documentation/rpmb/index.rst             |  18 ------
 Documentation/rpmb/introduction.rst      | 102 -------------------------------
 Documentation/rpmb/rpmb-tool.rst         |  19 ------
 Documentation/rpmb/simulation-device.rst |  19 ------
 MAINTAINERS                              |   1 -
 8 files changed, 167 deletions(-)
 delete mode 100644 Documentation/rpmb/conf.py
 delete mode 100644 Documentation/rpmb/index.rst
 delete mode 100644 Documentation/rpmb/introduction.rst
 delete mode 100644 Documentation/rpmb/rpmb-tool.rst
 delete mode 100644 Documentation/rpmb/simulation-device.rst

diff --git a/Documentation/conf.py b/Documentation/conf.py
index b9cadc9..63857d3 100644
--- a/Documentation/conf.py
+++ b/Documentation/conf.py
@@ -404,8 +404,6 @@ latex_documents = [
      'The kernel development community', 'manual'),
     ('userspace-api/index', 'userspace-api.tex', 'The Linux kernel user-space API guide',
      'The kernel development community', 'manual'),
-    ('rpmb/index', 'rpmb.tex', 'Linux RPMB Subsystem Documentation',
-     'The kernel development community', 'manual'),
 ]
 
 # The name of an image file (relative to this directory) to place at the top of
diff --git a/Documentation/index.rst b/Documentation/index.rst
index 380868d..cb7f1ba 100644
--- a/Documentation/index.rst
+++ b/Documentation/index.rst
@@ -75,7 +75,6 @@ needed).
    sound/index
    crypto/index
    filesystems/index
-   rpmb/index
 
 Architecture-specific documentation
 -----------------------------------
diff --git a/Documentation/rpmb/conf.py b/Documentation/rpmb/conf.py
deleted file mode 100644
index 15430a0..0000000
--- a/Documentation/rpmb/conf.py
+++ /dev/null
@@ -1,5 +0,0 @@
-# -*- coding: utf-8; mode: python -*-
-
-project = "Linux RPMB Subsystem"
-
-tags.add("subproject")
diff --git a/Documentation/rpmb/index.rst b/Documentation/rpmb/index.rst
deleted file mode 100644
index 876a260..0000000
--- a/Documentation/rpmb/index.rst
+++ /dev/null
@@ -1,18 +0,0 @@
-.. -*- coding: utf-8; mode: rst -*-
-
-==============================================
-Replay Protected Memory Block (RPMB) subsystem
-==============================================
-
-.. toctree::
-
-   introduction
-   simulation-device.rst
-   rpmb-tool.rst
-
-.. only::  subproject
-
-   Indices
-   =======
-
-   * :ref:`genindex`
diff --git a/Documentation/rpmb/introduction.rst b/Documentation/rpmb/introduction.rst
deleted file mode 100644
index f4ded7d..0000000
--- a/Documentation/rpmb/introduction.rst
+++ /dev/null
@@ -1,102 +0,0 @@
-.. -*- coding: utf-8; mode: rst -*-
-
-=============
-Introduction:
-=============
-
-Few storage technologies such is EMMC, UFS, and NVMe support RPMB
-hardware partition with common protocol and frame layout.
-The RPMB partition `cannot` be accessed via standard block layer,
-but by a set of specific commands:
-
-WRITE, READ, GET_WRITE_COUNTER, and PROGRAM_KEY.
-
-The commands and the data are embedded within :c:type:`rpmb_frame <rpmb_frame>`.
-
-An RPMB partition provides authenticated and replay protected access,
-hence it is suitable as a secure storage.
-
-In-kernel API
--------------
-The RPMB layer aims to provide in-kernel API for Trusted Execution
-Environment (TEE) devices that are capable to securely compute the block
-frame signature. In case a TEE device wish to store a replay protected
-data, it creates an RPMB frame with requested data and computes HMAC of
-the frame, then it requests the storage device via RPMB layer to store
-the data.
-
-The layer provides two APIs, for :c:func:`rpmb_req_cmd()` for issuing one of RPMB
-specific commands and :c:func:`rpmb_seq_cmd()` for issuing of raw RPMB protocol
-frames, which is close to the functionality provided by emmc multi ioctl
-interface.
-
-.. c:function:: int rpmb_cmd_req(struct rpmb_dev *rdev, struct rpmb_data *data);
-
-.. c:function:: int rpmb_cmd_seq(struct rpmb_dev *rdev, struct rpmb_cmd *cmds, u32 ncmds);
-
-
-A TEE driver can claim the RPMB interface, for example, via
-:c:func:`class_interface_register`:
-
-.. code-block:: c
-
-        struct class_interface tee_rpmb_intf = {
-                .class      = &rpmb_class;
-                .add_dev    = rpmb_add_device;
-                .remove_dev = rpmb_remove_device;
-        }
-        class_interface_register(&tee_rpmb_intf);
-
-
-RPMB device registeration
-----------------------------
-
-A storage device registers its RPMB hardware (eMMC) partition or RPMB
-W-LUN (UFS) with the RPMB layer :c:func:`rpmb_dev_register` providing
-an implementation for :c:func:`rpmb_seq_cmd()` handler. The interface
-enables sending sequence of RPMB standard frames.
-
-.. code-block:: c
-
-        struct rpmb_ops mmc_rpmb_dev_ops = {
-                .cmd_seq = mmc_blk_rpmb_cmd_seq,
-                .type = RPMB_TYPE_EMMC,
-                ...
-        }
-        rpmb_dev_register(disk_to_dev(part_md->disk), &mmc_rpmb_dev_ops);
-
-
-User space API
---------------
-
-A parallel user space API is provided via /dev/rpmbX character
-device with two IOCTL commands.
-Simplified one, ``RPMB_IOC_REQ_CMD``, were read result cycles is performed
-by the framework on behalf the user and second, ``RPMB_IOC_SEQ_CMD`` where
-the whole RPMB sequence, including ``RESULT_READ`` is supplied by the caller.
-The latter is intended for easier adjusting of the applications that
-use ``MMC_IOC_MULTI_CMD`` ioctl, such as
-https://android.googlesource.com/trusty/app/storage/
-
-.. code-block:: c
-
-        struct rpmb_ioc_req_cmd ireq;
-        int ret;
-
-        ireq.req_type = RPMB_WRITE_DATA;
-        rpmb_ioc_cmd_set(ireq.icmd, RPMB_F_WRITE, frames_in, cnt_in);
-        rpmb_ioc_cmd_set(ireq.ocmd, 0, frames_out, cnt_out);
-
-        ret = ioctl(fd, RPMB_IOC_REQ_CMD, &ireq);
-
-
-API
----
-.. kernel-doc:: include/linux/rpmb.h
-
-.. kernel-doc:: drivers/char/rpmb/core.c
-
-.. kernel-doc:: include/uapi/linux/rpmb.h
-
-.. kernel-doc:: drivers/char/rpmb/cdev.c
-
diff --git a/Documentation/rpmb/rpmb-tool.rst b/Documentation/rpmb/rpmb-tool.rst
deleted file mode 100644
index 148eb0d..0000000
--- a/Documentation/rpmb/rpmb-tool.rst
+++ /dev/null
@@ -1,19 +0,0 @@
-==========
-RPMB Tool
-==========
-
-There is a sample rpmb tool under tools/rpmb/ directory that exercises
-the RPMB devices via RPMB character devices interface (/dev/rpmbX)
-
-.. code-block:: none
-
-        rpmb [-v] [-r|-s] <command> <args>
-
-        rpmb program-key [-<RPMB_DEVICE> <KEY_FILE>
-        rpmb write-counter <RPMB_DEVICE> [KEY_FILE]
-        rpmb write-blocks <RPMB_DEVICE> <address> <block_count> <DATA_FILE> <KEY_FILE>
-        rpmb read-blocks <RPMB_DEVICE> <address> <blocks count> <OUTPUT_FILE> [KEY_FILE]
-
-        rpmb -v/--verbose:  runs in verbose mode
-        rpmb -s/--sequence: use RPMB_IOC_SEQ_CMD
-        rpmb -r/--request:  use RPMB_IOC_REQ_CMD
diff --git a/Documentation/rpmb/simulation-device.rst b/Documentation/rpmb/simulation-device.rst
deleted file mode 100644
index 9192a78..0000000
--- a/Documentation/rpmb/simulation-device.rst
+++ /dev/null
@@ -1,19 +0,0 @@
-======================
-RPMB Simulation Device
-======================
-
-RPMB partition simulation device is a virtual device that
-provides simulation of the RPMB protocol and uses kernel memory
-as storage.
-
-This driver cannot promise any real security, it is suitable for testing
-of the RPMB subsystem it self and mostly it was found useful for testing of
-RPMB applications prior to RPMB key provisioning/programming as
-The RPMB key programming can be performed only once in the life time
-of the storage device.
-
-Implementation:
----------------
-
-.. kernel-doc:: drivers/char/rpmb/rpmb_sim.c
-
diff --git a/MAINTAINERS b/MAINTAINERS
index 1ae34b0..d5328d5 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -11572,7 +11572,6 @@ F:	drivers/char/rpmb/*
 F:	include/uapi/linux/rpmb.h
 F:	include/linux/rpmb.h
 F:	Documentation/ABI/testing/sysfs-class-rpmb
-F:	Documentation/rpmb.rst
 F:	tools/rpmb/
 
 RTL2830 MEDIA DRIVER
-- 
2.7.4

